### PR TITLE
Add "wide URL bar" option

### LINKED ISSUE #8421: Offer option to have URL bar fill empty space ("wide URL bar")
## Test plan

https://github.com/brave/browser-laptop/pull/8714#issue-226475313

----

- Did you search for similar issues before submitting this one?
yes

- Describe the issue you encountered:

screenshot_20170421_144620

- Platform
KaOS, self build with this setup: https://gist.github.com/ShalokShalom/faf765bb82b3363a2215567523cd3ae6

- Brave Version (revision SHA):
3de60d52df3ef48d12d9696a076b72079beb9658

This here is the build log: 

https://gist.github.com/ShalokShalom/86e51be92dca5ec979445c695d56f95a

### PR DESCRIPTION
**Pull request by luixxiul**
Fix #8421

Auditors:

Test Plan 1:
1. Clear the browser profile
2. Open the browser
3. Open about:preferences#general
4. Make sure "Use wide URL bar" is disabled
5. Enable the option
6. Make sure the URL bar fills the margin around itself

Test Plan 2:
1. Open about:preferences#extensions
2. Enable LastPass extension
3. Make sure there is margin between the URL bar and the extension's icon

Test Plan 3:
1. Minimize the window's width
2. Make sure there does not appear extraDragArea between the URL bar and the extension's icon

- [x] Submitted a [ticket](https://github.com/brave/browser-laptop/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- [x] Ran `git rebase -i` to squash commits (if needed).

Test Plan:

### CODE REVIEW

File: app/extensions/brave/locales/en-US/preferences.properties [hunk 1]
```diff
@@ -325,6 +325,7 @@ alwaysDeny=Always deny
 alwaysAsk=Always ask
 autoHideMenuBar=Hide the menu bar by default
 disableTitleMode=Always show the URL bar
+wideURLbar=Use wide URL bar
```
Comment (luixxiul) [inline app/extensions/brave/locales/en-US/preferences.properties]: I actually am not sure if it sounds native. Should it be rather something like `Maximize the URL bar` or `Make the URL bar wide`, etc?

Comment (bsclifton) [inline app/extensions/brave/locales/en-US/preferences.properties]: @luixxiul I think that's a good name 😄 

Other names might be too complex, like "Have URL bar fill entire width" (which sounds confusing)

**Comment by cezaraugusto** [comment]
code looks good but I'm not sure if we want the UI as-is. Mind adding this commit to a branch in the main repo so @bradleyrichter can check it out as well?

**Comment by luixxiul** [comment]
np I'll upload gif later.

**Comment by luixxiul** [comment]
also it looks some media queries are working around 600px on about:preferences.

wideurlbar

it does not happen with normal pages.

**Comment by luixxiul** [comment]
Also I found the fix for the issue above. I'll push another commit.

**Comment by luixxiul** [comment]
wideurlbar

**Comment by luixxiul** [comment]
ready for re-review.

File: app/renderer/components/navigation/navigationBar.js [hunk 2]
```diff
@@ -167,6 +168,8 @@ class NavigationBar extends React.Component {
   }
 
   render () {
+    const wideURLbar = getSetting(settings.WIDE_URL_BAR)
```
Comment (NejcZdovc) [inline app/renderer/components/navigation/navigationBar.js]: All variables that are used in renderer needs to be defined in `mergeProps`

File: app/renderer/components/navigation/urlBar.js [hunk 3]
```diff
@@ -483,9 +483,12 @@ class UrlBar extends React.Component {
   }
 
   render () {
+    const wideURLbar = getSetting(settings.WIDE_URL_BAR)
```
Comment (NejcZdovc) [inline app/renderer/components/navigation/urlBar.js]: ditto

File: app/renderer/components/navigation/navigationBar.js (hunk 2 above)
Comment (luixxiul) [inline app/renderer/components/navigation/navigationBar.js]: Fixed with https://github.com/brave/browser-laptop/commit/9edeee43af5c8fbe733fab4baeb2d61fbc70b5a8

File: app/renderer/components/navigation/urlBar.js (hunk 3 above)
Comment (luixxiul) [inline app/renderer/components/navigation/urlBar.js]: Fixed with https://github.com/brave/browser-laptop/commit/9edeee43af5c8fbe733fab4baeb2d61fbc70b5a8

File: app/renderer/components/navigation/navigationBar.js (hunk 2 above)
Comment (NejcZdovc) [inline app/renderer/components/navigation/navigationBar.js]: perfect thank you

**Comment by bradleyrichter** [comment]
I only see one change needed for this PR:

Let's move both URL bar options from general prefs to advanced prefs. General is becoming heavy and should be reserved for the most basic options rather than overrides.

image

**Comment by luixxiul** [comment]
OK I'll address that.

**Comment by luixxiul** [comment]
Updated.

File: docs/state.md [hunk 4]
```diff
@@ -183,7 +183,8 @@ AppStore
     'extensions.pocket.enabled': boolean, // true if pocket is enabled
     'extensions.vimium.enabled': boolean, // true if vimium is enabled
     'general.autohide-menu': boolean, // true if the Windows menu should be autohidden
-    'general.bookmarks-toolbar-mode': boolean, // true to show bookmakrs toolbar
+    'general.wide-url-bar': boolean, // true to use wide URL bar
+    'general.bookmarks-toolbar-mode': boolean, // true to show bookmarks toolbar
```
Comment (luixxiul) [inline docs/state.md]: just fixed typo

**Comment by luixxiul** [comment]
Since the issue has been addressed, I'm going to merge this. Let's address other issues if any with follow-ups.

**Comment by bsclifton** [comment]
Great job, @luixxiul! This looks amazing. I'll definitely be using it 😄

**Pull request merged**